app/javascript/controllers/search_product_controller.js

```diff
@@ -0,0 +1,20 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["form"]
+
+  connect() {
+    this.timer = null;
+  }
+
+  submit() {
+    clearTimeout(this.timer);
+    this.timer = setTimeout(() => {
+      this.formTarget.requestSubmit();
+    }, 300);
+  }
+
+  get formTarget() {
+    return this.targets.find("form");
+  }
```

Comment:
I don't think this is needed 🤔 

https://stimulus.hotwired.dev/reference/targets